From 6b55afd691b6433ccfa8abb31bdb41e38b659cd0 Mon Sep 17 00:00:00 2001
From: David Sommerseth <davids@openvpn.net>
Date: Mon, 26 May 2025 16:54:09 +0200
Subject: [PATCH 4/5] addon/aws: Prepare for bumping the required C++ standard
 version to C++20

The openvpn3-service-aws feature was not included in the C++20 changes
in commit e7960a3c9e98af6b02d0782f88b7ff29bc07ea7c.

Signed-off-by: David Sommerseth <davids@openvpn.net>
---
 addons/aws/openvpn3-service-aws.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/addons/aws/openvpn3-service-aws.cpp b/addons/aws/openvpn3-service-aws.cpp
index 0e2946a..a2e9705 100644
--- a/addons/aws/openvpn3-service-aws.cpp
+++ b/addons/aws/openvpn3-service-aws.cpp
@@ -108,9 +108,9 @@ class AWSObject : public DBus::Object::Base
 
         log->LogInfo("Running on instance " + route_context->instance_id() + ", route table " + route_table_id);
 
-        subscr_mgr->Subscribe(signals_target, "NetworkChange", [=](DBus::Signals::Event::Ptr &event)
+        subscr_mgr->Subscribe(signals_target, "NetworkChange", [this](DBus::Signals::Event::Ptr &event)
                               {
-                                  process_network_change(event);
+                                  this->process_network_change(event);
                               });
 
         netcfg_mgr->NotificationSubscribe(NetCfgChangeType::ROUTE_ADDED | NetCfgChangeType::ROUTE_REMOVED);
-- 
2.49.0

